src/transformers/typescript.ts

```diff
@@ -152,7 +152,9 @@ function injectVarsToCode({
   // TODO investigate if it's possible to achieve this with a
   // TS transformer (previous attemps have failed)
   const codestores = Array.from(
-    contentForCodestores.match(/\$[a-zA-Z_][\w_$]*/g) || [],
+    contentForCodestores.match(
+      /\$[^\s();:,[\]{}.?!+\-=*/\\~|&%<>^`"'°§#1-9][^\s();:,[\]{}.?!+\-=*/\\~|&%<>^`"'°§#]*/g,
```

Comment:
You need the groups because while the variable can not start with a number, it can have a number after it